### DISCUSSION TITLE
Version tag vs released version mismatch

### DISCUSSION THREAD

**Discussion by PropmanProphecy** [microsoft/AL-Go#1350] [Q&A]
To be able to start the release process we need to fill out some required field regarding the Name and the Version tag for the release.

how do we "guess" what the next build version will be?
Since according to documentation Al-Go is in 100% complete control of the version process yes we can control the major and minor but not the build version and since it is required in the tag we need to guess on what it will be.

you can not just only add the first 2 number since the version needs to be semantic correct else the pipeline will fail.

or am I misunderstanding something.
 
![image](https://github.com/user-attachments/assets/6ec54dd6-4c34-4166-9465-f9ef1104f1f8)

Or am I misunderstanding something.

if not then the tag should be controlled be AL-Go the same way as the app version.

**Comment by freddydk** [comment]
Recommendation is to use major.minor.hotfix (where hotfix goes 0, 1, 2 etc) for the tag - I will add this to the docs (and window)

**Reply by frottke** [reply]
If I understand correctly, this is in connection to PR #1356, where you're also introducing the ability to set the build version segment via app.json and the "increment version number" workflow.

This would be another step towards achieving better compatibility with SemVer. Would it be also possible to use the fourth segment for the RunAttempt/RunNumber?

**Reply by freddydk** [reply]
In that PR is a test to use the fourth segment as runnumber and ignore runattempt.
That versioning strategy (3) might fit better to semver on the release site because the build number no longer part of the first 3 segments - and you basically manage 3 segments yourself.

**Reply by frottke** [reply]
perfect :)

It would be great if this functionality in the PR could become a feature in the future, we will certainly use it.

**Comment by freddydk** [comment]
Versioningstrategy 3 has been shipped with this functionality
